lxd/images.go

```diff
@@ -984,8 +1061,66 @@ func doDeleteImage(d *Daemon, fingerprint string) error {
 func imageDelete(d *Daemon, r *http.Request) Response {
 	fingerprint := mux.Vars(r)["fingerprint"]
 
+	deleteFromAllPools := func() error {
+		// Use the fingerprint we received in a LIKE query and use the full
+		// fingerprint we receive from the database in all further queries.
+		_, imgInfo, err := dbImageGet(d.db, fingerprint, false, false)
+		if err != nil {
+			return err
+		}
+
+		poolIDs, err := dbImageGetPools(d.db, imgInfo.Fingerprint)
+		if err != nil {
+			return err
+		}
+
+		if len(poolIDs) == 0 {
+			// Use the fingerprint we received in a LIKE query and use the full
+			// fingerprint we receive from the database in all further queries.
+			id, imgInfo, err := dbImageGet(d.db, fingerprint, false, false)
+			if err != nil {
+				return err
+			}
+
+			// Remove main image file.
+			fname := shared.VarPath("images", imgInfo.Fingerprint)
+			if shared.PathExists(fname) {
+				err = os.Remove(fname)
+				if err != nil {
+					shared.LogDebugf("Error deleting image file %s: %s", fname, err)
+				}
+			}
+
+			// Remove the rootfs file for the image.
+			fname = shared.VarPath("images", imgInfo.Fingerprint) + ".rootfs"
+			if shared.PathExists(fname) {
+				err = os.Remove(fname)
+				if err != nil {
+					shared.LogDebugf("Error deleting image file %s: %s", fname, err)
+				}
+			}
+
+			// Remove the database entry for the image.
+			return dbImageDelete(d.db, id)
+		}
```

Comment:
Yeah, this is a leftover from the prior implementation that was targeting pool specific image deletions.